lib/valueflow.cpp

```diff
@@ -3699,8 +3699,11 @@ static void valueFlowSymbolicInfer(const SymbolDatabase& symboldatabase, const S
             if (astIsFloat(tok->astOperand2(), false))
                 continue;
 
-            SymbolicInferModel leftModel{tok->astOperand1()};
-            std::vector<ValueFlow::Value> values = infer(leftModel, tok->str(), 0, tok->astOperand2()->values());
+            std::vector<ValueFlow::Value> values;
```

Comment:
I don't an opinion but we could also write:

     std::vector<ValueFlow::Value> values = [tok](){
            SymbolicInferModel leftModel{tok->astOperand1()};
            return infer(leftModel, tok->str(), 0, tok->astOperand2()->values());
     }();

Use the syntax you like..

Reply:
I do not find that more readable. I am also not sure fi that might introduce some overhead.

I mainly changed this because I was debugging the lifetime of the infer models at some point and was confused that two of them were alive at the same time.

Reply:
ok np